src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -33,7 +33,7 @@ public ServerSentEventsTransport(IReadableChannel<Message> application, ILoggerF
                 while (await _application.WaitToReadAsync(context.RequestAborted))
                 {
                     Message message;
-                    if (_application.TryRead(out message))
+                    while (_application.TryRead(out message))
```

Comment:
It does strike me that WaitToReadAsync could still avoid heap allocation when a wait isn't needed since the return is a bool so it could cache the completed `true` and `false` tasks. Still, better for us to avoid depending on that and do the loop ourselves as well.